### PR TITLE
Different return type in ca expression list

### PR DESCRIPTION
**Pull request by jirimosinger**

### CODE REVIEW

File: parser_library/test/expressions/ca_expr_list_test.cpp [hunk 1]
```diff
@@ -176,6 +178,31 @@ TEST(ca_expr_list, unfinished_expressions)
...
+{
+    // ((1 AND 1)+1)
+    context::hlasm_context ctx;
+    diagnostic_op_consumer_container diags;
+    evaluation_context eval_ctx { ctx, library_info_transitional::empty, diags };
+
+    std::vector<ca_expr_ptr> inner_list;
+    inner_list.emplace_back(std::make_unique<ca_constant>(1, range()));
+    inner_list.emplace_back(std::make_unique<ca_constant>(1, range()));
+    inner_list.emplace_back(std::make_unique<ca_symbol>(context::id_index("AND"), range()));
```
Comment (slavek-kucera) [inline parser_library/test/expressions/ca_expr_list_test.cpp:195]: This looks more like `1 1 AND`

File: parser_library/src/expressions/conditional_assembly/terms/ca_expr_list.cpp [hunk 2]
```diff
@@ -73,11 +73,7 @@ void ca_expr_list::apply(ca_expr_visitor& visitor) const { visitor.visit(*this);
 
 context::SET_t ca_expr_list::evaluate(const evaluation_context& eval_ctx) const
 {
-    assert(expr_list.size() <= 1);
-
-    if (expr_list.empty())
-        return context::SET_t(expr_kind);
-    return expr_list.front()->evaluate(eval_ctx);
+    return expr_list.size() == 1 ? expr_list.front()->evaluate(eval_ctx) : context::SET_t(expr_kind);
```
Comment (slavek-kucera) [inline parser_library/src/expressions/conditional_assembly/terms/ca_expr_list.cpp:76]: Was the assert triggered?

Comment (jirimosinger) [reply parser_library/src/expressions/conditional_assembly/terms/ca_expr_list.cpp:76]: It was triggered in debug mode for something like this:
```
         AIF   ( (1 AND 1).AAAA).NAMERR
```

Comment (slavek-kucera) [reply parser_library/src/expressions/conditional_assembly/terms/ca_expr_list.cpp:76]: That would indicate that the resolve_expression/resolve_expression_tree was not called somewhere...

**Review by slavek-kucera — APPROVED** [review]
Maybe run the fuzzer for a while?

**Pull request merged**